src/components/Signup/Signup.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { 
+  ChangeEvent, 
+  FormEvent, 
+  useState 
+} from "react";
+
+import "./Signup.css";
+
+const Signup = () => {
+  const formState = {
+    email: '',
+    password: ''
+  };
+
+  const [form, setForm] = useState(formState);
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    if (name == "email") {
+      setForm( current => {
+        return { ...current, email: value };
+      });
+    } else {
+      setForm( current => {
+        return { ...current, password: value };
+      });
+    }
+  };
```

Comment:
Remove the conditional checks. And you can use the `setForm` in one go when you set the `name` from `event.target` as a key:
```js
setForm(prevState => ({
  ...prevState,
  [name]: event.target.value
});
```
This makes the `handleChange` event handler more dynamic and less verbose when dealing with multiple input elements.

Reply:
Some feedback -- `form` variable never used.  Is there a way to show the form data `onSubmit` in the console?